src/connection.rs

```diff
@@ -1302,6 +1302,108 @@ where
     }
 }
 
+async fn send_error_response<'a, R: AsyncRead, W: AsyncWrite>(
+    mut resp: server::Response<'_, R, W>,
+    zreceiver: &TrackedAsyncLocalReceiver<'_, (arena::Rc<zhttppacket::OwnedResponse>, usize)>,
+    e: &Error,
+) -> Result<(), Error> {
+    let headers = &[http1::Header {
+        name: "Content-Type",
+        value: b"text/plain",
+    }];
+
+    let mut body: ArrayVec<u8, 512> = ArrayVec::new();
+
+    let code = match e {
+        Error::CoreHttp(CoreHttpError::Http(e)) => {
+            writeln!(&mut body, "Failed to parse request: {}", e)?;
+
+            400
```

Comment:
can we  use `http::status::StatusCode` constants 
https://docs.rs/http/latest/http/status/struct.StatusCode.html

Reply:
I don't think I'd add that crate right now just for this, but it is an interesting crate we may want to consider later